### DISCUSSION TITLE
Include shortcode not rendering inline R code

### DISCUSSION THREAD

**Discussion by bbartholdy** [quarto-dev/quarto-cli#931] [Q&A] [answered]
Hi,

I have a quarto document (let's call it *file1.qmd*) where I would like to include a second quarto document (*_file2.qmd*) that contains the following text and inline R code:

```
This is a file to test the execution of includes inline code.

Inline code test: `r version$version.string`
```

In *file1.qmd* I have the shortcode `{{< include _file2.qmd >}}`

When I render *file1.qmd* with quarto (v. 0.9.426) it currently looks like:

>This is a file to test the execution of includes inline code.
>
>Inline code test: `r version$version.string`

I would like it to look like:

>This is a file to test the execution of includes inline code.
>
>Inline code test: R version 4.2.0 (2022-04-22)

Am I doing something wrong or is it currently not possible to include files with inline R code and have the inline code executed?

**Comment by jjallaire** [comment]
It's probable that we aren't detecting that the knitr engine is in play (b/c we don't see any R chunks or inline R expressions in the main document). You can add `engine: knitr` to make sure that this binding occurs: https://quarto.org/docs/computations/execution-options.html#engine-binding

**Reply by cscheid** [reply]
I bet that's what's happening there. I'll create a github issue to track, because we _are_ attempting to redetect the engines after include expansion, so if this is the case, then it's a bug on my code.

**Reply by bbartholdy** [reply]
Indeed! Simply setting `engine: knitr` at the top of the .qmd file fixed it. Thanks!

**Comment by cscheid** [comment, marked as answer]
This is now fixed on https://github.com/quarto-dev/quarto-cli/commit/19021eb80298eef313507722eb5a0ca4d9780108. Thanks for reporting!
